File: output/anthropic_claude_sonnet_4_6/zero/python/Cálculo.py

def solve():
    line1 = input().split()
    M, N = int(line1[0]), int(line1[1])
    X = list(map(int, input().split()))
    Y = list(map(int, input().split()))
    
    max_len = max(M, N)
    
    # Pad both to same length
    X_padded = X + [0] * (max_len - M)
    Y_padded = Y + [0] * (max_len - N)
    
    # Add from right to left
    result = []
    carry = 0
    for i in range(max_len - 1, -1, -1):
        s = X_padded[i] + Y_padded[i] + carry
        result.append(s % 2)
        carry = s // 2
    
    # If carry remains, prepend (but problem says X+Y < 1, so carry at position 0 means integer part)
    # Actually carry here would mean the sum >= 1, but constraints say X+Y < 1
    # Still handle it just in case
    if carry:
        result.append(carry)
    
    result.reverse()
    
    # Remove trailing zeros (obi-2015 uses minimum digits, last digit must be 1)
    while len(result) > 1 and result[-1] == 0:
        result.pop()
    
    print(' '.join(map(str, result)))

solve()

File: output/anthropic_claude_sonnet_4_6/few/cpp/Cálculo.cpp
#include <bits/stdc++.h>
using namespace std;
int main(){
    ios::sync_with_stdio(0);
    cin.tie(0);
    int M, N;
    cin >> M >> N;
    vector<int> X(M), Y(N);
    for(int i=0;i<M;i++) cin>>X[i];
    for(int i=0;i<N;i++) cin>>Y[i];
    int maxLen = max(M,N);
    // pad to same length
    vector<int> A(maxLen,0), B(maxLen,0);
    for(int i=0;i<M;i++) A[i]=X[i];
    for(int i=0;i<N;i++) B[i]=Y[i];
    // add from right to left
    vector<int> res(maxLen,0);
    int carry=0;
    for(int i=maxLen-1;i>=0;i--){
        int s=A[i]+B[i]+carry;
        res[i]=s%2;
        carry=s/2;
    }
    // carry should be 0 since X+Y<1
    // remove trailing zeros
    int last=maxLen-1;
    while(last>0 && res[last]==0) last--;
    for(int i=0;i<=last;i++){
        if(i>0) cout<<" ";
        cout<<res[i];
    }
    cout<<endl;
    return 0;
}